src/app/api/admin/[...slug]/route.js

```diff
@@ -1,44 +1,53 @@
 import { defaultHandler } from 'ra-data-simple-prisma';
 import { NextResponse } from 'next/server';
 import { prisma } from '@/lib/db';
-import { MODEL_INCLUDES, searchInputFilters, transformServiceProvider, withErrorHandlerAndAuth } from '../common';
+import { auth } from '@/lib/auth';
+import { withErrorHandler } from '@/lib/errors/errorHandler';
+import { NotAuthorizedException } from '@/lib/errors/NotAuthorizedException';
+import { MODEL_INCLUDES, searchInputFilters, transformServiceProvider } from '../common';
 
-const handler = withErrorHandlerAndAuth(async req => {
-  const json = await req.json();
+const handler = auth(
+  withErrorHandler(async req => {
+    if (!req.auth) throw new NotAuthorizedException();
 
-  const modelName = json.resource.toLowerCase();
-  const isServiceProvider = modelName === 'specialist' || modelName === 'organization';
+    const json = await req.json();
 
-  const getOneTransform = instance => {
-    transformServiceProvider(instance, modelName);
-  };
+    const modelName = json.resource.charAt(0).toLowerCase() + json.resource.slice(1);
```

Comment:
🟢  consider using _.camelCase maybe? :)

Reply:
Thanks for comment! Discussed in private conversation.